File: 0101-0200/0169-Majority Element/0169-Majority Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        candidate = nums[0]
        for num in nums:
            if num == candidate:
                count += 1
            else:
                count -= 1
                if count == 0:
                    count = 1
                    candidate = num
        return candidate

File: 0101-0200/0169-Majority Element/0169-Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0, candidate = nums[0];
        for (int num : nums) {
            if (num == candidate) {
                ++count;
            }
            else {
                if (--count == 0) {
                    candidate = num;
                    count = 1;
                }
            }
        }
        
        return candidate;
    }
};
